Add ACCESS_ONCE() macro, extracted from the following commit upstream:

    From 11c397c207dda26084be8a5da5bd78c74b4b3651 Mon Sep 17 00:00:00 2001
    From: Andrew Cooper <andrew.cooper3@citrix.com>
    Date: Wed, 7 Dec 2016 13:52:02 +0000
    Subject: [PATCH] x86/time: Adjust init-time handling of pit0_ticks

---
 xen/include/xen/lib.h  | 2 ++

diff --git a/xen/include/xen/lib.h b/xen/include/xen/lib.h
index d1171b774b13..1976e4be8dcb 100644
--- a/xen/include/xen/lib.h
+++ b/xen/include/xen/lib.h
@@ -56,6 +56,8 @@
 
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]) + __must_be_array(x))
 
+#define ACCESS_ONCE(x) (*(volatile typeof(x) *)&(x))
+
 #define MASK_EXTR(v, m) (((v) & (m)) / ((m) & -(m)))
 #define MASK_INSR(v, m) (((v) * ((m) & -(m))) & (m))
 
-- 
2.26.3

